src/main/java/org/openstreetmap/atlas/generator/AtlasGeneratorHelper.java

```diff
@@ -115,12 +117,31 @@
                     final File temporaryLocalFile = File
                             .temporary(getAtlasName(country, shard) + "-", ATLAS_EXTENSION);
 
-                    System.out.println("Downloaded atlas from " + path
-                            + " and is found as temp file " + temporaryLocalFile.getAbsolutePath());
+                    logger.info("Downloaded atlas from {} and is found as temp file {}", path,
+                            temporaryLocalFile.getAbsolutePath());
 
                     // FileSystemHelper.resource sets the Decompressor on the Resource for us, so
                     // this call will gunzip the file
-                    fileFromNetwork.copyTo(temporaryLocalFile);
+                    try
+                    {
+                        fileFromNetwork.copyTo(temporaryLocalFile);
+                    }
+                    catch (final Exception e)
+                    {
+                        final Optional<FileNotFoundException> fileNotFound = ExceptionSearch
```

Comment:
Quick question Mike. Is there an advantage to using this `Optional` approach versus just doing something like:

```Java
try
{
    // code
}
catch (FileNotFoundException exception)
{
    // FileNotFoundException handling here
}
catch(Exception exception)
{
    // anything else falls down to here
}
```

I am curious if the `Optional` approach is providing some additional functionality that the above approach misses. Thanks!

Reply:
Great callout @lucaspcram! I wish we could do that, that would be a lot more straightforward. The reason for using `ExceptionSearch` here is that the `FileNotFoundException` happens at the very bottom of the stack - Hadoop FileSystem default behavior. That's then wrapped by a number of CoreExceptions in our code - the full stack trace had 3 or 4 layers. So we would have to walk the entire stack and find the root cause of the Exception and determine if it was FileNotFoundException and handle it accordingly. This is what `ExceptionSearch` is doing here, courtesy of @cstaylor.

Reply:
Ah wow ok that's hairy. I checked out the `ExceptionSearch` class. Very clever @cstaylor